### PR TITLE
[bugfix] USH-2211 incomplete form with EOF nav looses app ID

### PR DESCRIPTION
**Pull request by snopoke**
## Technical Summary
When completing an 'incomplete' form in Web Apps that has EOF nav, the very next screen is correct but navigation after that fails since the App ID has been lost.

This PR re-sets the app ID in the URL based on the response for formplayer (only if the URL does not have an appId param already).

Jira: https://dimagi-dev.atlassian.net/browse/USH-2211

### Safety story
Fairly small changes that are easy to follow and tested locally.

### Automated test coverage
None

### QA Plan
None

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js [hunk 1]
```diff
@@ -75,7 +85,7 @@ hqDefine("cloudcare/js/formplayer/menus/collections", function () {
             } else if (response.tree) {
                 // form entry time, doggy
                 _.extend(this, _.pick(response, this.formProperties));
-                FormplayerFrontend.trigger('startForm', response, this.app_id);
+                FormplayerFrontend.trigger('startForm', response, this.appId);
```
Comment (snopoke) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js:88]: This is unrelated to the bugfix.

This was referencing a non-existent var so it was always passing `undefined`. It doesn't seem to be used at all in the `startForm` handler so could potentially be removed.

Comment (snopoke) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/collections.js:88]:
```suggestion
                FormplayerFrontend.trigger('startForm', response);
```

**Review by orangejenny — APPROVED** [review]
What would you think of doing the replacement somewhere more generic, like adding a `done` handler to the promise created [here](https://github.com/dimagi/commcare-hq/blob/57ef26349702f2f8eeec370eefbf14d4d5b02286/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js#L148)?

I'm not totally sure it would be better - I don't think that the order of done callbacks is guaranteed, so any other code dealing with the response would need to not depend on app id being in the URL.

**Comment by snopoke** [comment]
> What would you think of doing the replacement somewhere more generic, like adding a `done` handler to the promise created [here](https://github.com/dimagi/commcare-hq/blob/57ef26349702f2f8eeec370eefbf14d4d5b02286/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js#L148)?
> 
> I'm not totally sure it would be better - I don't think that the order of done callbacks is guaranteed, so any other code dealing with the response would need to not depend on app id being in the URL.

I think it would be great to standardize how the URL get's updated based on a response from FP. There are a number of other places that already adjust the URL in various ways but I can try out your suggestion.
